File: src/main/java/net/atos/client/zgw/shared/exception/ValidatieFoutExceptionMapper.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.shared.exception;

import jakarta.ws.rs.core.MultivaluedMap;
import jakarta.ws.rs.core.Response;

import org.eclipse.microprofile.rest.client.ext.ResponseExceptionMapper;

import net.atos.client.zgw.shared.model.ValidatieFout;

/**
 *
 */
public class ValidatieFoutExceptionMapper
    implements ResponseExceptionMapper<ValidatieFoutException> {

  @Override
  public boolean handles(final int status, final MultivaluedMap<String, Object> headers) {
    return status == Response.Status.BAD_REQUEST.getStatusCode();
  }

  @Override
  public ValidatieFoutException toThrowable(final Response response) {
    return new ValidatieFoutException(response.readEntity(ValidatieFout.class));
  }
}

File: src/main/kotlin/nl/info/zac/shared/exception/ValidatieFoutExceptionMapper.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2024 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.shared.exception

import jakarta.ws.rs.core.Response
import jakarta.ws.rs.ext.ExceptionMapper
import jakarta.ws.rs.ext.Provider
import net.atos.client.zgw.shared.exception.ValidationErrorException
import nl.info.zac.app.exception.RestExceptionResponseBuilder.generateResponse
import nl.info.zac.exception.ErrorCode

/**
 * Exception mapper to catch [ValidationErrorException] thrown and convert them to a [Response] which is returned to
 * the frontend for further processing.
 */
@Provider
class ValidatieFoutExceptionMapper : ExceptionMapper<ValidationErrorException> {
    override fun toResponse(validationErrorException: ValidationErrorException): Response {
        return generateResponse(
            responseStatus = Response.Status.BAD_REQUEST,
            errorCode = ErrorCode.ERROR_CODE_VALIDATION_GENERIC,
            exception = validationErrorException,
            exceptionMessage = validationErrorException.validatieFout.invalidParams.toString()
        )
    }
}
